velox/functions/remote/client/Remote.cpp

```diff
@@ -71,7 +90,41 @@ class RemoteFunction : public exec::VectorFunction {
   }
 
  private:
-  void applyRemote(
+  void applyRestRemote(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const {
+    try {
+      serializer::presto::PrestoVectorSerde serde;
+      auto remoteRowVector = std::make_shared<RowVector>(
+          context.pool(),
+          remoteInputType_,
+          BufferPtr{},
+          rows.end(),
+          std::move(args));
+
+      std::unique_ptr<folly::IOBuf> requestBody =
```

Comment:
This is serializing all the rows in one go. So this is assuming a certain contract from the rest server exposing the function. We should make this clear.